spec/tbd_tests_spec.rb

```diff
@@ -4869,7 +5155,267 @@
+    material3.setName("poor material m tbd")
+    expect(material3.nameString).to eq("poor material m tbd")
+    expect(material3.setThermalResistance(1.0)).to be(true)
+    expect(material3.thermalResistance).to be_within(0.1).of(1.0)
+    mat3 = OpenStudio::Model::MaterialVector.new
+    mat3 << material3
+
+    # Assign highly-conductive material to a new construction.
+    construction = OpenStudio::Model::Construction.new(os_model)
+    construction.setName("poor construction")
+    expect(construction.nameString).to eq("poor construction")
+    expect(construction.layers.size).to eq(0)
+    expect(construction.setLayers(mat2)).to be(true) # or switch with 'mat'
+    expect(construction.layers.size).to eq(1)
+
+    # Assign " tbd" massless material to a new construction.
+    construction2 = OpenStudio::Model::Construction.new(os_model)
+    construction2.setName("poor construction tbd")
+    expect(construction2.nameString).to eq("poor construction tbd")
+    expect(construction2.layers.size).to eq(0)
+    expect(construction2.setLayers(mat3)).to be(true)
+    expect(construction2.layers.size).to eq(1)
+
+    # Assign construction to the "Office Left Wall".
+    expect(left_office_wall.setConstruction(construction)).to be(true)
+
+    # Assign construction2 to the "Fine Storage Right Wall".
+    expect(right_fine_wall.setConstruction(construction2)).to be(true)
+
+    subs = front_office_wall.subSurfaces
+    expect(subs.empty?).to be(false)
+    expect(subs.size).to eq(4)
+
+    psi_set = "poor (BETBG)"
+    ioP = File.dirname(__FILE__) + "/../json/tbd_warehouse9.json"
+    # {
+    #   "schema": "https://github.com/rd2/tbd/blob/master/tbd.schema.json",
+    #   "description": "testing error detection",
+    #   "psis": [
+    #     {
+    #       "id": "detailed_2",
+    #       "fenestration": 0.600
+    #     },
+    #     {
+    #       "id": "regular (BETBG)",                    <<<< non-fatal ERROR # 1
+    #       "fenestration": 0.700
+    #     }
+    #   ],
+    #   "khis": [
+    #     {
+    #       "id": "cantilevered beam",
+    #       "point": 0.6
+    #     }
+    #   ],
+    #   "surfaces": [
+    #     {
+    #       "id": "Office Front Wall",
+    #       "khis": [
+    #         {
+    #           "id": "beam",                           <<<< non-fatal ERROR # 2
+    #           "count": 3
+    #         }
+    #       ]
+    #     },
+    #     {
+    #       "id": "Office Left Wall",
+    #       "khis": [
+    #         {
+    #           "id": "cantilevered beam",
+    #           "count": 300                           <<<< triggers WARNING # 1
+    #         }
+    #       ]
+    #     }
+    #   ],
+    #   "edges": [
+    #     {
+    #       "psi": "detailed",                          <<<< non-fatal ERROR # 3
+    #       "type": "fenestration",
+    #       "surfaces": [
+    #         "Office Front Wall",
+    #         "Office Front Wall Window 1"
+    #       ]
+    #     }
+    #   ]
+    # }
```

Comment:
In addition to the preceding OSM anomalies introduced earlier (e.g., borderline RSi values, 5-sided windows), pointing out here the invalid TBD JSON entries that will trigger warnings or errors.